### PR TITLE
Add hint support for search request

### PR DESCRIPTION
**Pull request by leZudiste**
Vertx mongo client doesn't support the hint() helper method and we need it to optimize our queries by using specific index.

### CODE REVIEW

File: src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java [hunk 1]
```diff
@@ -975,6 +975,9 @@ private JsonObject decodeKeyWhenUseObjectId(JsonObject json) {
     if (options.getFields() != null) {
       find.projection(wrap(options.getFields()));
     }
+    if (!options.getHint().isEmpty()) {
```
Comment (vietj) [inline src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java:978]: can you check nullity ?

**Review by vietj — CHANGES_REQUESTED** [review]
minor change

File: src/main/generated/io/vertx/ext/mongo/FindOptionsConverter.java [hunk 2]
```diff
@@ -59,5 +64,8 @@ public static void toJson(FindOptions obj, java.util.Map<String, Object> json) {
     if (obj.getSort() != null) {
       json.put("sort", obj.getSort());
     }
+    if (!obj.getHint().isEmpty()) {
```
Comment (tsegismont) [inline src/main/generated/io/vertx/ext/mongo/FindOptionsConverter.java:67]: It looks like you have modified this file manually. In this case, can you please run `mvn compile` and commit the generated files instead? Thank you

**Comment by leZudiste** [comment]
Other modifications are necessary?

**Comment by vietj** [comment]
thanks for the contribution

**Pull request merged**